### PR TITLE
Fix port web process

### LINKED ISSUE #15: Providing a port based on the procfile command is broken
`if (command[0] == "web" ...`

This doesn't work because by the time we get to this point, `command` has already been replaced with the contents of the procfile.

Need to save the original command in a separate variable (or something)

### PR DESCRIPTION
**Pull request by j4mie**
Fixes #15 and adds tests around port-providing behaviour.

### CODE REVIEW

File: tests/test_cli.py [hunk 1]
```diff
@@ -38,3 +38,26 @@ def test_has_updated_path(self):
...
+    def test_port_provided_if_explicitly_requested(self):
+        os.environ["CRAB_PROVIDE_PORT"] = "true"
+        main(["somecommand"])
+        provided_port = self.execvpe.call_args[0][2].get("PORT")
+        self.assertTrue(provided_port and provided_port.isdigit())
+        del os.environ["CRAB_PROVIDE_PORT"]
+
+    def test_port_not_provided_by_default(self):
+        main(["somecommand"])
+        self.assertTrue("PORT" not in self.execvpe.call_args[0][2])
```
Comment (RealOrangeOne) [inline tests/test_cli.py:63]:
```suggestion
        self.assertNotIn("PORT", self.execvpe.call_args[0][2])
```

File: tests/test_cli.py [hunk 2]
```diff
@@ -38,3 +38,26 @@ def test_has_updated_path(self):
...
+    def test_port_provided_for_command_containing_port(self):
+        main(["somecommand", "$PORT"])
+        provided_port = self.execvpe.call_args[0][2].get("PORT")
+        self.assertTrue(provided_port and provided_port.isdigit())
+
+    def test_port_provided_if_explicitly_requested(self):
+        os.environ["CRAB_PROVIDE_PORT"] = "true"
+        main(["somecommand"])
+        provided_port = self.execvpe.call_args[0][2].get("PORT")
+        self.assertTrue(provided_port and provided_port.isdigit())
```
Comment (RealOrangeOne) [inline tests/test_cli.py:58]:
```suggestion
        self.assertTrue(provided_port.isdigit())
```

File: tests/test_cli.py [hunk 3]
```diff
@@ -38,3 +38,26 @@ def test_has_updated_path(self):
...
+        os.environ["PROC_FILE"] = "tests/Procfile"
+        main(["web"])
+        provided_port = self.execvpe.call_args[0][2].get("PORT")
+        self.assertTrue(provided_port and provided_port.isdigit())
+        del os.environ["PROC_FILE"]
+
+    def test_port_provided_for_command_containing_port(self):
+        main(["somecommand", "$PORT"])
+        provided_port = self.execvpe.call_args[0][2].get("PORT")
+        self.assertTrue(provided_port and provided_port.isdigit())
```
Comment (RealOrangeOne) [inline tests/test_cli.py:52]:
```suggestion
        self.assertTrue(provided_port.isdigit())
```

File: tests/test_cli.py [hunk 4]
```diff
@@ -38,3 +38,26 @@ def test_has_updated_path(self):
         path = self.execvpe.call_args[0][2]["PATH"].split(":")
         self.assertEqual(path[0], "env/bin")
         self.assertEqual(path[1], os.getcwd())
+
+    def test_port_provided_for_web_command_in_procfile(self):
+        os.environ["PROC_FILE"] = "tests/Procfile"
+        main(["web"])
+        provided_port = self.execvpe.call_args[0][2].get("PORT")
+        self.assertTrue(provided_port and provided_port.isdigit())
```
Comment (RealOrangeOne) [inline tests/test_cli.py:46]:
```suggestion
        self.assertTrue(provided_port.isdigit())
```

File: tests/test_cli.py (hunk 2 above)
Comment (j4mie) [reply tests/test_cli.py:58]: `provided_port` can be `None` so this check needs to be there.

**Pull request merged**